atc/component/scheduler.go

```diff
@@ -0,0 +1,76 @@
+package component
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"code.cloudfoundry.org/clock"
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/lager/lagerctx"
+)
+
+var Clock = clock.NewClock()
```

Comment:
Is there a reason to make this a global variable? It could make testing more difficult.

Reply:
It's actually exposed so that it can be reassigned in the tests. 🙂 I don't feel too strongly about this but I've seen that pattern in other projects and it feels like a decent way to keep objects from ending up "over-injected" with things that we'd never want to change IRL.

Reply:
Do the testify tests not run in parallel?

I feel like "over-injected" methods is a smell that we shouldn't be hiding away.

Reply:
I definitely don't plan to do that for *all* injection; it's mainly that telling an object to use a "real clock" in production code when it's clearly just for testing feels a little goofy and adds noise to the real live code which we'll be reading forever.

`go test` runs tests in series unless you explicitly call `t.Parallel()` in the tests you want to run in parallel. (Side note: this looks like a good thing to watch to suss out any order dependence - https://github.com/golang/go/issues/28592)